File: packages/typescript-sdk/src/agent.ts
import type { AgentTestResponse } from "./types.js";
import type { GeneralAugmentClient } from "./client.js";

export class AgentClient {
  private client: GeneralAugmentClient;
  private projectId: string;

  constructor(client: GeneralAugmentClient, projectId: string) {
    this.client = client;
    this.projectId = projectId;
  }

  async test(
    message: string,
    options: { phoneE164?: string; channel?: string } = {}
  ): Promise<AgentTestResponse> {
    return this.client.testAgent(this.projectId, message, options);
  }
}

export async function testAgent(
  client: GeneralAugmentClient,
  projectId: string,
  message: string,
  options: { phoneE164?: string; channel?: string } = {}
): Promise<AgentTestResponse> {
  return client.testAgent(projectId, message, options);
}

File: packages/python-sdk/src/genaug/agent.py
"""Agent test helpers for General Augment SDK users."""

from __future__ import annotations

from typing import Any

from genaug.client import GeneralAugmentClient


class AgentClient:
    """Convenience wrapper scoped to one General Augment project."""

    def __init__(self, client: GeneralAugmentClient, project_id: str) -> None:
        """Initialize a project-scoped agent client."""
        self.client = client
        self.project_id = project_id

    def test(
        self,
        message: str,
        *,
        phone_e164: str = "+15550000000",
        channel: str = "whatsapp",
    ) -> dict[str, Any]:
        """Send a test message to the configured agent."""
        return self.client.test_agent(
            self.project_id,
            message,
            phone_e164=phone_e164,
            channel=channel,
        )


def test(
    client: GeneralAugmentClient,
    project_id: str,
    message: str,
    *,
    phone_e164: str = "+15550000000",
    channel: str = "whatsapp",
) -> dict[str, Any]:
    """Send a one-off test message to a General Augment project."""
    return client.test_agent(
        project_id,
        message,
        phone_e164=phone_e164,
        channel=channel,
    )
